app/controllers/api/base_controller/renderer.rb

```diff
@@ -540,6 +546,45 @@ def render_options(resource, data = {})
         klass = collection_class(resource)
         render :json => OptionsSerializer.new(klass, data).serialize
       end
+
+      def virtual_attributes_for(klass)
+        return nil unless klass.respond_to?(:reflect_on_association)
+
+        type    = @req.subject
+        results = []
+
+        validate_attr_selection(klass).last.each do |vattr|
+          next if vattr == "href_slug"
+
+          attr_name, attr_base = split_virtual_attribute(vattr)
+          filtered_attr        = yield type, attr_name, attr_base
```

Comment:
The block returning filtered_attr is kind of confusing as that's not what I'd expect from a method named `virtual_attributes_for`, but I can't think of something better at the moment. 😕

Reply:
@Fryguy Would maybe just adding a comment explaining how this method functions be a good middle ground for now?

Reply:
Sounds good

Reply:
Done.